packages/typespec-vscode/src/vscode-cmd/create-tsp-project.ts

```diff
@@ -300,146 +325,114 @@ async function selectEmitters(
   return Object.fromEntries(selectedEmitters.map((x) => [x.name, x.emitterTemplate]));
 }
 
-async function tspInstall(
-  client: TspLanguageClient,
-  directory: string,
-): Promise<ExecOutput | undefined> {
-  logger.info("Installing TypeSpec project dependencies by 'tsp install'...");
-  return await vscode.window.withProgress(
-    {
-      location: vscode.ProgressLocation.Notification,
-      title: "Installing TypeSpec project dependencies by 'tsp install'...",
-      cancellable: true,
-    },
-    async (_progress, token) => {
-      const TIMEOUT = 600000; // set timeout to 10 minutes which should be enough for tsp install for a new project
-      try {
-        const result = await createPromiseWithCancelAndTimeout(
-          client.runCliCommand(["install"], directory),
-          token,
-          TIMEOUT,
-        );
-        return result;
-      } catch (e) {
-        if (e === ResultCode.Cancelled) {
-          logger.info(
-            "Installation of TypeSpec project dependencies by 'tsp install' is cancelled by user",
-          );
-          return undefined;
-        } else if (e === ResultCode.Timeout) {
-          logger.error(
-            `Installation of TypeSpec project dependencies by 'tsp install' is timeout after ${TIMEOUT}ms`,
-          );
-          return undefined;
+async function installDependencies(directory: string): Promise<Result<ExecOutput | undefined>> {
+  logger.info("Installing project dependencies");
+  const checkTspCliPromise = checkInstalledTspCli();
+  const checkNpmPromise = checkInstalledNpm();
+  const [checkTspCli, checkNpm] = await Promise.all([checkTspCliPromise, checkNpmPromise]);
+  if (checkTspCli || checkNpm) {
+    return await tryExecuteWithUi(
+      {
+        name: "Installing project dependencies",
+        progress: {
+          timeoutInMs: 5 * 60 * 1000, // 5 minutes
+          title: "Installing project dependencies",
+          withCancelAndTimeout: true,
+        },
+      },
+      async (progress) => {
+        if (checkTspCli) {
+          logger.info("tsp cli is installed, try to install dependencies with tsp install");
```

Comment:
This is an improvement over the current behavior as it should now support the standalone but it still has the same issues that this issue describe https://github.com/microsoft/typespec/issues/6321

Not really the scope of this PR but just wanted to make sure this issue wasn't as well marked as resolved from this pr.

Reply:
yeah, this PR won't resolve #6321 and we will check whether we can refactor and bundle the ```tsp install``` logic directly which will be ideal when working on #6321. thx